### PR TITLE
Fixes for issues 3191, 3222, 3411

### PR DESCRIPTION
**Pull request by tony-gaia**
Issue date not longer settable on form, this is set when the licence is issued (not when saved) and will always be set to the current date at time of issue.

Relevant initial licence field values are copied from previous licence in the case of amend/renew (excluding end_date on renewals).

An applications is_temporary property is determined only by the customer's temp status.

Applications that are temporary will not show up on any dashboard.

A licence's start date must now be no earlier than the current date.

Fixed problem with selecting a variant on select licence type causing the unexpected leave warning message / session application deletion to occur.

Licences that have been saved but not issued will now show in the dashboard as "Unissued" and the action will be "Issue", which takes them to the issue licence page.

### CODE REVIEW

**Comment by tony-gaia** [comment]
Also fixes 3287

File: wildlifelicensing/apps/applications/views/issue.py [hunk 1]
```diff
@@ -145,17 +149,33 @@ def get_context_data(self, **kwargs):
...
+                licence_initial_data['regions'] = previous_licence.regions.all().values_list('pk', flat=True)
+                licence_initial_data['locations'] = previous_licence.locations
+                licence_initial_data['purpose'] = previous_licence.purpose
+                licence_initial_data['additional_information'] = previous_licence.additional_information
+                licence_initial_data['cover_letter_message'] = previous_licence.cover_letter_message
+
+                if application.is_licence_amendment:
+                    licence_initial_data['end_date'] = previous_licence.end_date
+
+            if not 'purpose' in licence_initial_data or len(licence_initial_data['purpose']) == 0:
```
Comment (serge-gaia) [inline wildlifelicensing/apps/applications/views/issue.py]: PEP8 recommends `not in`

File: wildlifelicensing/apps/applications/views/issue.py [hunk 2]
```diff
@@ -145,17 +149,33 @@ def get_context_data(self, **kwargs):
...
+                licence_initial_data['return_frequency'] = application.licence_type.returntype.month_frequency
             else:
-                return_frequency = -1
+                licence_initial_data['return_frequency'] = -1
 
-            kwargs['issue_licence_form'] = IssueLicenceForm(purpose=purposes,
-                                                            is_renewable=application.licence_type.is_renewable,
-                                                            default_period=application.licence_type.default_period,
-                                                            return_frequency=return_frequency)
+            kwargs['issue_licence_form'] = IssueLicenceForm(**licence_initial_data)
```
Comment (serge-gaia) [inline wildlifelicensing/apps/applications/views/issue.py:178]: Should it not be  
```
form = IssueLicenceForm(initial=licence_initial_data)
```
instead? By setting the form parameter `initial` you shouldn't have to do your own initialisation (see below)

File: wildlifelicensing/apps/main/forms.py [hunk 3]
```diff
@@ -88,13 +87,15 @@ class Meta:
...
+        end_date = kwargs.pop('end_date', None)
         is_renewable = kwargs.pop('is_renewable', False)
-
         default_period = kwargs.pop('default_period', None)
-
         return_frequency = kwargs.pop('return_frequency', WildlifeLicence.DEFAULT_FREQUENCY)
+        locations = kwargs.pop('locations', '')
+        purpose = kwargs.pop('purpose', '')
+        additional_information = kwargs.pop('additional_information', '')
+        cover_letter_message = kwargs.pop('cover_letter_message', '')
```
Comment (serge-gaia) [inline wildlifelicensing/apps/main/forms.py:98]: with the `initial` form parameter (see above) you could avoid all this, right?

File: wildlifelicensing/apps/main/forms.py [hunk 4]
```diff
@@ -105,34 +106,41 @@ def __init__(self, *args, **kwargs):
...
+            self.fields['locations'].initial = locations
+            self.fields['purpose'].initial = purpose
+            self.fields['additional_information'].initial = additional_information
+            self.fields['cover_letter_message'].initial = cover_letter_message
+
+    def clean_start_date(self):
+        start_date = self.cleaned_data['start_date']
+        if start_date < date.today():
+            raise forms.ValidationError('Start date cannot before current date')
+
```
Comment (serge-gaia) [inline wildlifelicensing/apps/main/forms.py]: This rule is only for new licences, not amended or reissued (see Walter [comment](https://kanboard.dpaw.wa.gov.au/?controller=TaskViewController&action=show&task_id=3411&project_id=24))

**Pull request merged**